lightning-invoice/src/payment.rs

```diff
@@ -25,10 +25,12 @@ use lightning::routing::router::{PaymentParameters, RouteParameters};
 ///
 /// Will always succeed unless the invoice has an amount specified, in which case
 /// [`payment_parameters_from_invoice`] should be used.
+/// An amount-less invoice, also known as a zero-amount invoice, refers to an invoice
+/// where the amount to be paid is zero. In this context, both terms are used synonymously.
```

Comment:
I think the added docs are conveying the wrong message here.
A zero-amount invoice doesn't mean the amount to be paid is zero, but rather any arbitrary amount can be paid by the sender. For context, [this](https://bitcoin.stackexchange.com/a/111722) answer might be helpful.

Also, FWIW, we don't need to explain the meaning of zero-amount invoice here, as the users of this function would most probably be aware of it.
That's why, I think a simple indication that **zero-amount ≈ amountless** will suffice here.

Reply:
Got it will change the docs accordingly